gcc/rust/checks/errors/rust-feature-gate.cc

```diff
@@ -21,6 +21,27 @@
 
 namespace Rust {
 
+const std::map<std::string, FeatureGate::GatedBuiltinAttriSymbol>
+  FeatureGate::gated_builtin_attri_symbol_map = {
+    {"rustc_builtin_macro",
```

Comment:
All rustc_* attributes are feature gated. Even those unknown by the compiler.

Reply:
The number of them is nearly 40+ (rustc 1.49).

Reply:
I mean you should check for any attribute starting with `rustc_` and apply the feature gate rather than hard coding a specific set of known rustc\_\* attributes.

Reply:
This `symbol -> attri` map is just to define the diagnostic string for each rustc_* attributes. If I delete this map, I need to write these  strings in a very long `switch` statment.

Reply:
You can use `internal implementation detail` as diagnostic string for all rustc_* attributes.

Reply:
I don't known. Does this messages should be aligned with rustc? If not, I can delete this `gated_builtin_attri_symbol_map`.

Reply:
IMHO error messages don't need to match between gccrs and rustc. It doesn't match between different versions of rustc either. There is some value in getting error messages at least somewhat close to rustc, but I don't think it matters at all for internal things like rustc_* attributes.